### ISSUE TITLE
Propose to give droided side control to newly connected users?

### ISSUE THREAD

**Issue by vgaming** [wesnoth/wesnoth#2597]
I just saw the following game on MP server: two players played a game seemingly OK. Then one of them disconnected, and re-connected back. Both did not know the (complex?) algo to give control back, so they quit the game and never played it to the end.

Can we do something about such a story line? (No, don't propose the all-explaining clip from Microsoft Office and Silicon Valley TV show:)

I imagine one of the following could work:
* if you droided a side because of a player disconnection, propose to offer the side to any new observer once it connects

Any other alternatives? Thoughts?

**Comment by ProditorMagnus** [comment]
I suggested in #2500 control/give_control - needs GUI access as well for 1.14

**Comment by vgaming** [comment]
@ProditorMagnus note that this issue proposes to _offer side_ by itself, without any user actions at all. This is a different from the goal from making `give_control` simpler.
So these feature requests are not directly dependent.

**Comment by sapientN3T** [comment]
if it is the same nick as the one who disconnected, sure. other than that, it is considered rude to give control to an observer before checking with them first. so for etiquette reasons I would not encourage that. consider also that the observer may need to catch up on the replay for a long time, so the other players would be stuck waiting a long time in that scenario.

**Comment by vgaming** [comment]
@sapientN3T good point about the confusion if you get a side you didn't expect to get. With that in mind, I'd change the proposal to give control to the same nickname only (ideally it would be checking that one nick is prefix/suffix of the other, but that would be too hard to explain/understand, so I'm not proposing this).

**Comment by amokmen** [comment]
There is only one right decision. Like in one Quake 2 multiplayer teamplay mod - when match starts every player is given some **join code** (smth like cookies in Internet browsers). And if player disconnects with any reason (power off, Internet off, game hanged) - he could reconnect to game server and use his **join code** to retake his position in team.

**Comment by GregoryLundberg** [comment]
Using a session cookie is certainly a solution. As with all solutions, there are problems. Most likely, no solution solves all the problems.

**Comment by AI0867** [comment]
There's actually a provision for session identifiers in the network code, to allow clients to reclaim their previous connection. It was never actually implemented though, probably because of security considerations.

Another approach would be to automatically give the side back to a connecting player if the player has the same registered username.

**Comment by sapientN3T** [comment]
why implement for registered only? realistically, if the host sees the same nick return they will pass control back without a question 100%. So if the player was worried about impersonation then they coulda-woulda-shoulda registered.

However since the old nick is often hanging around as a ghost, it can prevent them from using their nick. This cookie could be used to bump the ghost.

**Comment by ProditorMagnus** [comment]
>  if the host sees the same nick return they will pass control back without a question

Often the question would be how to change control.
